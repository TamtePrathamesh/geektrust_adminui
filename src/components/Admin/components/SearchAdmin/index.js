import React from 'react';
import PropTypes from 'prop-types';

import Style from './style.module.css';

const SearchAdmin = ({ onFilter }) => {
  return (
    <>
      <input
        data-testid="search"
        className={Style.search}
        type="text"
        placeholder="Search by name, email or role"
        onChange={onFilter}
      />
    </>
  );
};

SearchAdmin.propTypes = {
  onFilter: PropTypes.func,
};

export default React.memo(SearchAdmin);
