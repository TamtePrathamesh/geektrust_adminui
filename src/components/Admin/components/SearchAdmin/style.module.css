@import '../../../../style/colors.css';

.search {
  width: 100%;
  height: 40px;
  border: 1px solid var(--grey_one);
  border-radius: 4px;
  padding-left: 10px;
  letter-spacing: 2px;
}

.search:focus-visible {
  border: 2px solid var(--grey_one);
}

@media screen and (max-width: 750px) {
  .search {
    width: 100%;
    height: 40px;
    border: 1px solid var(--grey_one);
    border-radius: 4px;
  }
}
