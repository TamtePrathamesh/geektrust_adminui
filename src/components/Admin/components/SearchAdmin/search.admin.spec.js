import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import SearchAdmin from '.';

describe('SearchAdmin', () => {
  it('should match snapshot', () => {
    const onFilter = jest.fn();
    expect(render(<SearchAdmin onFilter={onFilter} />).container).toMatchSnapshot();
  });

  it('should call filter on input text changed', () => {
    const onFilter = jest.fn();
    render(<SearchAdmin onFilter={onFilter} />);
    expect(onFilter).toHaveBeenCalledTimes(0);
    user.type(screen.queryByTestId('search'), 'admin');
    expect(onFilter).not.toHaveBeenCalledTimes(0);
  });
});
