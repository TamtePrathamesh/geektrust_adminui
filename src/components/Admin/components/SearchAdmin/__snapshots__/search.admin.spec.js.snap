// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAdmin should match snapshot 1`] = `
<div>
  <input
    class="search"
    data-testid="search"
    placeholder="Search by name, email or role"
    type="text"
  />
</div>
`;
